lib/fs.js

```diff
@@ -156,8 +165,11 @@ function makeStatsCallback(cb) {
 
 function nullCheck(path, callback) {
   if (('' + path).indexOf('\u0000') !== -1) {
-    var er = new Error('Path must be a string without null bytes');
-    er.code = 'ENOENT';
+    const er = new errors.Error('ERR_INVALID_ARG_TYPE',
```

Comment:
IMHO this could be changed to `TypeError`.
Feedback anyone?

Reply:
That's really difficult to say. The coercion to a string means that any value can be used, it's not really a `TypeError`. The real check here is the null-character bit, which doesn't *feel* like a `TypeError` to me. I could live with it tho.